lua/ide/components/explorer/filenode.lua

```diff
@@ -95,6 +95,11 @@ FileNode.new = function(path, kind, perms, depth)
             table.insert(children, fnode)
         end
         log.debug("found %d children", #children)
+				if list_directories_first then
+						table.sort(children, function(first, second)
+						  return first ~= second and first.kind == 'dir' and second.kind ~= 'dir'
```

Comment:
How do you expect that 

```
first ~= second
```

Clause to work? 

Do we actually need to call "vim.tbl_deep_equal" or do a 

```
first.key ~= second.key
```

They keys are meant to be unique, so its probably good enough.

Reply:
It only matters because of the way sort works in Lua. We just need to check referential quality so that it’s a valid sorting function.

if you return `true` when the items are deferentially equal, you get an error about invalid sorting function.

If both items are directories, sort should be unchanged. Functionally, the `first ~= second` is not really doing anything but it’s needed for Lua’s sort implementation.

Reply:
Yeah, this is required behavior for Lua sorting functions. https://www.lua.org/pil/19.3.html

Reply:
cool, TIL.